modules/terraform-aws-vpc/data.tf

```diff
@@ -16,3 +16,23 @@
 data "aws_availability_zones" "zones_in_region" {
   state = "available"
 }
+
+# Find details of the AWS vpc
+data "aws_vpc" "vpc" {
+  id = local.vpc_id
+}
+
+data "aws_subnets" "vpc_subnets" {
+  filter {
+    name   = "vpc-id"
+    values = [local.vpc_id]
+  }
+
+  # Postcondition to verify subnets are part of VPC
+  lifecycle {
+    postcondition {
+      condition     = (length(setsubtract(local.public_subnet_ids, self.ids)) == 0) && (length(setsubtract(local.private_subnet_ids, self.ids)) == 0)
+      error_message = "One or more of the provided subnets - ${join(",", setsubtract(concat(local.public_subnet_ids, local.private_subnet_ids), self.ids))} - are not part of VPC ${local.vpc_id}"
```

Comment:
@jimright We will likely have a few more VPC-related validations (elastic IP quota&limits check if create\_vpc = true). On the other hand, those validations will be more policy-based, whereas checking whether the provided subnets are part of the right VPC is indeed a very basic check of correctness. Therefore, while I agree that such basic validations as this one belong here directly in the code, could we put them into a central place in the module, so that they are more visible? If that is not possible in TF, we can of course just leave it as-is.

A different but related question:
How/where will policy-based VPC validations (e.g. connectivity requirement check, subnet size check, elastic IP limit check) be implemented?

Reply:
> Could we put them into a central place in the module, so that they are more visible? If that is not possible in TF, we can of course just leave it as-is.

Yes, we should talk some more about the design for the validation but my plan for the more in-depth validation checks (the quota checks; connectivity, etc.) is to have these in a separate validation module. This might also see the small number of checks we already have moving to that module.
Also, Open Policy Agent that we have been discussing might have different requirements for placement/definition of these checks.